app/models/katello/candlepin/content.rb

```diff
@@ -79,6 +79,11 @@ def build_repository
         repository
       end
 
+      def get_certificate_and_key(product, modified_product_ids = [])
+        product = Product.where(:cp_id => modified_product_ids[0]).first if modified_product_ids.any?
+        {:cert => product.certificate, :key => product.key}
+      end
+
```

Comment:
I'm passing in ```product``` from above, which I assume from the code before is assumed to have a value. line 83 will just assign a new Product (from the modified product id) to ```product```